pkg/limayaml/defaults.go

```diff
@@ -297,7 +297,11 @@ func FillDefault(y, d, o *LimaYAML, filePath string) {
 	mounts := make([]Mount, 0, len(d.Mounts)+len(y.Mounts)+len(o.Mounts))
 	location := make(map[string]int)
 	for _, mount := range append(append(d.Mounts, y.Mounts...), o.Mounts...) {
+		if mount.SSHFS.FollowSymlinks == nil {
+			mount.SSHFS.FollowSymlinks = pointer.Bool(false)
+		}
```

Comment:
Setting the default cannot happen here; you need to merge from all the sources first, and then have another loop over the final result, filling in the missing values with their defaults. Look at how the `networks` are being merged, and copy that mechanism.

Adding test cases for this to `defaults_test.go` would help to show that the current logic does not work as expected.